Clarified TE scope requirements for GitLab SSO

cwarnermm opened the pull request:
Documentation for: https://github.com/mattermost/mattermost-server/issues/17499#issuecomment-978814072
- Clarified scope requirements for Mattermost Team Edition vs. all other MM subscription plans.
- Updated Mattermost config settings to update the config file with scope.
- Specified that updating the Mattermost config file requires a server restart.

cwarnermm commented on source/onboard/sso-gitlab.rst:
```diff
@@ -50,7 +50,10 @@ Step 1: Add an OpenID Connect application to your GitLab account
-  c. Select the scopes: ``openid``, ``profile``, and ``email``.
+  c. Select scopes.
+  
+     - For Mattermost Team Edition, select "read_user".
```
@sbishel - Do we want to specify the Mattermost releases this applies to? Based on the original issue, this may have been a change between ~v5.30 and v5.34?

sbishel replied:
Pretty sure this applies to all editions as Team Edition only supports Gitlab OAuth.

coltoneshaw commented on source/onboard/sso-gitlab.rst:
```diff
@@ -50,7 +50,10 @@ Step 1: Add an OpenID Connect application to your GitLab account
-  c. Select the scopes: ``openid``, ``profile``, and ``email``.
+  c. Select scopes.
+  
+     - For Mattermost Team Edition, select "read_user".
```
@cwarnermm - I apologize for this possibly naive question but what makes the team's implementation of OpenID different than Enterprise? I _believe_ that `OpenID` scope would be required for all of the versions.

cwarnermm replied:
@coltoneshaw - Some replies to the original thread suggested that ``OpenID`` scope wasn't working as expected. @sbishel - Can you help clarify, please?

sbishel replied:
Unless this has changed...OpenId is only available via Enterprise. Team Edition should still be supporting GitLab via OAuth. We don't allow Team Edition to have OpenId, because then they can use it to connect to anything.

coltoneshaw replied:
Are we continuing support for OAuth then? I thought it was being deprecated entirely, or already was.

coltoneshaw replied:
I _do_ think if we continue this then we'll want to break the docs apart or add a section since this doc focuses specifically on OpenID.

sbishel replied:
No, we never deprecated OAuth. We only modified the UI and only for Enterprise. OpenID is simply a "generic" OAuth with "well known" endpoints, the backend still supports OAuth.

coltoneshaw replied:
@cwarnermm - We may want to update the docs about OpenID then. It includes the below and doesn't include any steps related to OAuth 2.0 in this documentation.

> OAuth 2.0 is being deprecated and replaced by OpenID Connect. Refer to product documentation to convert your existing OAuth configuration for GitLab to the OpenID Connect standard.

oguera replied:
Hey! Just found this conversation and i want to support @coltoneshaw statement. In fact i created an issue earlier this week specific for this phrasing: https://github.com/mattermost/docs/issues/5338

coltoneshaw commented on source/onboard/sso-gitlab.rst:
```diff
@@ -65,9 +68,11 @@ Step 2: Configure Mattermost for GitLab SSO
 5. Paste the **Application ID** from GitLab as the **Client ID** in Mattermost.
 6. Paste the **Application Secret Key** from GitLab as the **Client Secret** in Mattermost. 
-7. Select **Save**.
+7. Update the ``config.json`` file to specify the scopes selected for the ``GitLabSettings`` property. Changes to this setting require a server restart before taking effect.
```
Based on the other feedback above, we may need to call out that `openid` is a required scope at a minimum for the connector to work.

mkraft replied:
Should we provide an example to clarify whether this is a space or comma separated list, or is this obvious enough?

coltoneshaw commented:
@cwarnermm - Actually, reading through the other ticket I see my original confusion. It seems that some people are still using the Gitlab connector to be OAuth 2.0 instead of OpenID. _If_ we want to continue OAuth 2.0 in the docs we may need to make it clear that this specific docs page pertains to OpenID specifically as we deprecated OAuth 2.0

cwarnermm commented:
@wiersgallak - Based on our recent conversations about improving the clarity of our SSO offerings in our product docs, I've made the following updates in this PR:
- [GitLab SSO](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/5317/onboard/sso-gitlab.html)
   - Removed the **All Plans** badge which was introducing confusion given that we support different standards depending on the subscription plan
   - Removed Migration section about OAuth 2.0 being deprecated in favor of OpenID Connect (since Engineering has confirmed that the deprecation has not moved forward)
   - Clarified in the first note that TE supports OAuth 2.0, and paid plans support OpenID Connect
- [Google SSO](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/5317/onboard/sso-google.html), [Office SSO](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/5317/onboard/sso-office.html), and [Convert OAuth 2.0 to OpenID Connect](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/5317/onboard/convert-oauth20-service-providers-to-openidconnect.html)
   - Removed Migration details

Outstanding: Remove the in-product notice that OAuth 2.0 is being deprecated in favor or Open ID Connect. This work will be addressed in a separate PR.

wiersgallak reviewed (approved):
Thanks, Carrie!
